### PR TITLE
docs: update release.md with improvements from v3.4.34 release

### PR DESCRIPTION
**Pull request by ivanvc**
This pull request has the improvements noted while doing #18486.

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

File: Documentation/contributor-guide/release.md [hunk 1]
```diff
@@ -78,11 +78,14 @@ which don't need to be executed before releasing each version.
...
    It generates all release binaries under the directory `/tmp/etcd-release-${VERSION}/etcd/release/` and images. Binaries are pushed to the Google Cloud bucket
    under project `etcd-development`, and images are pushed to `quay.io` and `gcr.io`.
+
+   **Remove the `quay.io` login entry from `~/.docker/config.json` after pushing Docker images.**
 7. Publish the release page on GitHub
    - Set the release title as the version name
    - Choose the correct release tag (generated from step #4)
    - Follow the format of previous release pages
    - Attach the generated binaries and signature file
+   - Verify the historical binary size for each architecture. If there's a big difference, verify that it works for that architecture
```
Comment (jmhbnz) [inline Documentation/contributor-guide/release.md:88]: I think longer term we could have a workflow trigger either automatically or manually during a release window which performs artifact validations.  

For now let's just add this note about manual verification though.

**Review by jmhbnz — APPROVED** [review]
LGTM - Thanks @ivanvc

File: Documentation/contributor-guide/release.md (hunk 1 above)
Comment (serathius) [reply Documentation/contributor-guide/release.md:88]: I would love to run e2e tests on the generated artifacts.

File: Documentation/contributor-guide/release.md [hunk 2]
```diff
@@ -78,11 +78,14 @@ which don't need to be executed before releasing each version.
 
    It generates all release binaries under the directory `/tmp/etcd-release-${VERSION}/etcd/release/` and images. Binaries are pushed to the Google Cloud bucket
    under project `etcd-development`, and images are pushed to `quay.io` and `gcr.io`.
+
+   **Remove the `quay.io` login entry from `~/.docker/config.json` after pushing Docker images.**
```
Comment (serathius) [inline Documentation/contributor-guide/release.md:82]: Didn't see what's the current way of procuring quay password. Let's make sure we persist it somewhere before we instruct everyone to delete. As detailed in https://contribute.cncf.io/resources/project-services/faq/#how-do-i-share-credentials-passwords-or-other-confidential-information there are free plans from password managers for open source projects. Would be great to adopt them.

Comment (ivanvc) [reply Documentation/contributor-guide/release.md:82]: @serathius, I think the issue is that no current maintainer can access the quay.io admin account. We've been using a shared password that @jmhbnz provides before the release, but we don't have a way to restrict the user's permissions, so the current safest option is to delete the login from the Docker config file.

File: Documentation/contributor-guide/release.md (hunk 1 above)
Comment (ivanvc) [reply Documentation/contributor-guide/release.md:88]: I completely agree. This is a short-term solution while we implement better automation.

File: Documentation/contributor-guide/release.md (hunk 2 above)
Comment (jmhbnz) [reply Documentation/contributor-guide/release.md:82]: Putting the change in this pr to the side I agree as a project we should implement some project level password vault so we are not reliant on credentials being shared directly by maintainers as that assumes maintainers are always available when needed and always remember / have their own system to store the passwords.

Comment (ahrtr) [reply Documentation/contributor-guide/release.md:82]: Looks like [1Password’s free open source plan](https://github.com/1Password/for-open-source) is a good choice to proceed.

**Review by ahrtr — APPROVED** [review]
LGTM

We can resolve the followups separately.

File: Documentation/contributor-guide/release.md (hunk 2 above)
Comment (ahrtr) [reply Documentation/contributor-guide/release.md:82]: Raised https://github.com/etcd-io/etcd/issues/18611

**Pull request merged**